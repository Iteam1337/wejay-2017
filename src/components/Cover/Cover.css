/* ==================================================================
   Album cover
   ================================================================== */
.Cover {
  position: relative; }

.Cover__image {
  border-radius: 4px;
  max-width: 100%;
  position: relative;
  vertical-align: top;
  z-index: 2; }

.Cover__shadow {
  bottom: -25px;
  filter: blur(15px);
  left: 0;
  max-width: 100%;
  position: absolute;
  transform: scale(0.9);
  z-index: 1; }

.Cover__shadow--small {
  bottom: -10px;
  transform: scale(0.7); }

.Cover__temp {
  background: #ccc;
  border-radius: 4px; }

.Position {
  height: 310px;
  left: -5px;
  position: absolute;
  top: -5px;
  width: 310px; }

.shape {
  stroke-dasharray: 1500;
  stroke-dashoffset: 1500;
  stroke-linecap: round;
  stroke-linejoin: round;
  stroke-width: 5px;
  fill: transparent;
  stroke: rgba(255, 134, 0, 0.4);
  transition: stroke-dashoffset 100ms ease-in-out; }
