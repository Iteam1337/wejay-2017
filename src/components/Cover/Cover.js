// @flow

import * as WejayApi from '__generated__/types.flow'
import React from 'react'
import LazyLoad from 'react-lazyload'
import styled from 'styled-components'

type CoverProps = {
  track: WejayApi.TrackInfoFragment,
  width?: number,
}

const CoverImage = styled.img`
  border-radius: 2px;
  height: ${({ size }) => `${size}px`};
  max-width: 100%;
  vertical-align: top;
  width: ${({ size }) => `${size}px`};
`

const TemporaryCover = CoverImage.extend`
  background-color: #eaecef;
`

export const Cover = ({ track, width = 100 }: CoverProps) => {
  if (!track.album.images || track.album.images.length === 0) {
    return <TemporaryCover size={width} />
  }

  const covers = track.album.images
  const coverUrl = covers ? covers[0].url : ''

  return (
    <LazyLoad height={width}>
      <CoverImage alt="Album cover" size={width} src={coverUrl} />
    </LazyLoad>
  )
}

export default Cover
