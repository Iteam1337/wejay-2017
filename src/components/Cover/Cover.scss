/* ==================================================================
   Album cover
   ================================================================== */

.Cover {
  position: relative;
}

.Cover__image {
  border-radius: 4px;
  max-width: 100%;
  position: relative;
  vertical-align: top;
  z-index: 2;
}

.Cover__shadow {
  bottom: -25px;
  filter: blur(15px);
  left: 0;
  max-width: 100%;
  position: absolute;
  transform: scale(0.9);
  z-index: 1;
}

.Cover__shadow--small {
  bottom: -10px;
  transform: scale(0.7);
}

.Cover__temp {
  background: #333;
  border-radius: 4px;
}

.Position {
  bottom: 0;
  left: 0;
  height: 5px;
  position: fixed;
  right: 0;
}

.Position__bar {
  background: rgba(255, 134, 0, 0.4);
  border-radius: 0 2px 0 0;
  height: 5px;
  transition: width 100ms ease-in-out;
}
