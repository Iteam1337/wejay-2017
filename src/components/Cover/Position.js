import React, { Component } from 'react'

class Position extends Component {
  state = {
    position: 1500
  }

  timer = null

  componentDidMount () {
    this.startTimer(this.props)
  }

  componentWillReceiveProps (nextProps) {
    console.log(this.props.track, nextProps.track)
    if (this.props.track.spotifyUri !== nextProps.track.spotifyUri) {
      clearInterval(this.timer)

      this.startTimer(nextProps)
    }
  }

  startTimer = props => {
    const { track } = props

    this.timer = setInterval(() => {
      this.setState(() => ({
        position: 1500 - (Date.now() - track.started) / track.duration * 1500
      }))
    }, 1000)
  }

  render () {
    const { track } = this.props
    const style = {
      height: 310 / track.album.images[0].width * track.album.images[0].height,
      width: 310 / track.album.images[0].width * track.album.images[0].width
    }

    return (
      <div className="Position" style={style}>
        <svg
          height="310"
          width="310"
          xmlns="http://www.w3.org/2000/svg"
          style={style}
        >
          <rect
            className="shape"
            height="310"
            width="310"
            style={{
              ...style,
              strokeDashoffset: this.state.position
            }}
          />
        </svg>
      </div>
    )
  }
}

export default Position
