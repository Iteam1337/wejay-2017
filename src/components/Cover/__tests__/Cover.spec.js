import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Cover from '../Cover'

describe('components/Cover', () => {
  let component

  beforeEach(() => {
    component = shallow(
      <Cover
        album={{
          images: [
            {
              url: 'test.png'
            }
          ]
        }}
        width={50}
      />
    )
  })

  it('renders a temp cover if no images are provided', () => {
    component.setProps({
      album: {}
    })

    expect(toJson(component)).toMatchSnapshot()
  })

  it('renders Cover', () => {
    expect(toJson(component)).toMatchSnapshot()
  })
})
