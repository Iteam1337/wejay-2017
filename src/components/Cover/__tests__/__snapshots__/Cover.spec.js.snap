// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Cover renders Cover 1`] = `
<div
  className="Cover"
>
  <img
    alt="Album cover"
    className="Cover__image"
    src="test.png"
    style={
      Object {
        "width": 50,
      }
    }
  />
</div>
`;

exports[`components/Cover renders a temp cover if no images are provided 1`] = `
<div
  className="Cover__temp"
  style={
    Object {
      "height": 50,
      "width": 50,
    }
  }
/>
`;
